src/backend/dx12/Cargo.toml

```diff
@@ -13,6 +13,7 @@ workspace = "../../.."
 
 [features]
 default = ["winit"]
+single-threaded = ["gfx-hal/single-threaded"]
```

Comment:
why do `dx12`/`empty`/other non-GL backends need to expose the feature?

Reply:
Although uncommon but you may want to have all backends available (gl included). So you would need `gfx-hal` with `single-threaded` feature enabled. This will require all backends to use `single-threaded` feature.

Reply:
Enabling a feature for a base crate (gfx-hal in this case) doesn't force any dependencies to specify that. Feature is supposed to be additive, so cargo assumes everything will still compile (otherwise your `single-threaded` feature in the backends wouldn't be helping anyway).

Reply:
... still there